Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm

```diff
@@ -339,122 +318,117 @@
         }
     } else if (auto *fileExtension = path.pathExtension; fileExtension.length)
         result = [UTType typeWithFilenameExtension:fileExtension];
-    else if (auto *fileURL = resourceFileURLForPath(path))
+    else if (auto *fileURL = static_cast<NSURL *>(resourceFileURLForPath(path)))
         [fileURL getResourceValue:&result forKey:NSURLContentTypeKey error:nil];
 
     return result;
 }
 
-NSString *WebExtension::resourceStringForPath(NSString *path, RefPtr<API::Error>& outError, CacheResult cacheResult, SuppressNotFoundErrors suppressErrors)
-{
-    ASSERT(path);
-
-    // Remove leading slash to normalize the path for lookup/storage in the cache dictionary.
-    if ([path hasPrefix:@"/"])
-        path = [path substringFromIndex:1];
-
-    if (NSString *cachedString = objectForKey<NSString>(m_resources, path))
-        return cachedString;
-
-    if ([path isEqualToString:generatedBackgroundPageFilename] || [path isEqualToString:generatedBackgroundServiceWorkerFilename])
-        return generatedBackgroundContent();
-
-    NSData *data = resourceDataForPath(path, outError, CacheResult::No, suppressErrors);
-    if (!data)
-        return nil;
-
-    NSString *string;
-    [NSString stringEncodingForData:data encodingOptions:nil convertedString:&string usedLossyConversion:nil];
-    if (!string)
-        return nil;
-
-    if (cacheResult == CacheResult::Yes) {
-        if (!m_resources)
-            m_resources = [NSMutableDictionary dictionary];
-        [m_resources setObject:string forKey:path];
-    }
-
-    return string;
-}
-
-NSData *WebExtension::resourceDataForPath(NSString *path, RefPtr<API::Error>& outError, CacheResult cacheResult, SuppressNotFoundErrors suppressErrors)
+std::span<const uint8_t> WebExtension::resourceDataForPath(const String& path, RefPtr<API::Error>& outError, CacheResult cacheResult, SuppressNotFoundErrors suppressErrors)
 {
     ASSERT(path);
 
     outError = nullptr;
 
-    if ([path hasPrefix:@"data:"]) {
-        if (auto base64Range = [path rangeOfString:@";base64,"]; base64Range.location != NSNotFound) {
-            auto *base64String = [path substringFromIndex:NSMaxRange(base64Range)];
-            return [[NSData alloc] initWithBase64EncodedString:base64String options:0];
+    auto *cocoaPath = static_cast<NSString *>(path);
+
+    if ([cocoaPath hasPrefix:@"data:"]) {
+        if (auto base64Range = [cocoaPath rangeOfString:@";base64,"]; base64Range.location != NSNotFound) {
+            auto *base64String = [cocoaPath substringFromIndex:NSMaxRange(base64Range)];
+            auto data = retainPtr([[NSData alloc] initWithBase64EncodedString:base64String options:0]);
```

Comment:
No need for `RetainPtr` here, since we use ObjC ARC.
```suggestion
            auto *data = [[NSData alloc] initWithBase64EncodedString:base64String options:0];
```